### PR TITLE
Fix bug login with google

### PR DESCRIPTION
**Pull request by RWEMAREMY**
###what PR do?
fixing login with google

### CODE REVIEW

**Review by musabehonore — APPROVED** [review]

File: package.json [hunk 1]
```diff
@@ -116,6 +116,7 @@
     "express": "^4.19.2",
     "express-handlebars": "^7.1.2",
     "express-session": "^1.18.0",
+    "generate-password": "^1.7.1",
```
Comment (fab-ryan) [inline package.json]: Please don't use any extra packages generate random string for now

File: src/config/passport-setup.ts [hunk 2]
```diff
@@ -13,12 +13,14 @@ const googleStrategyOptions: StrategyOptionsWithRequest = {
...
   passReqToCallback: true,
 };
+
 passport.use(
   new GoogleStrategy(
     googleStrategyOptions,
     async (req, accessToken, refreshToken, profile, done) => {
       try {
         if (!profile.emails || profile.emails.length === 0) {
+          console.error('Email not found in the Google profile');
```
Comment (fab-ryan) [inline src/config/passport-setup.ts]: Remove this console

File: src/config/passport-setup.ts [hunk 3]
```diff
@@ -62,7 +65,7 @@ passport.serializeUser((user: unknown, done) => {
 
 passport.deserializeUser(async (id: string, done) => {
   try {
-    const user = await User.findByPk(id);
+    const user = await Database.User.findByPk(id);
```
Comment (fab-ryan) [inline src/config/passport-setup.ts]: use FindByOne not by PK

File: src/controllers/authController.ts [hunk 4]
```diff
@@ -29,22 +45,59 @@ export const handleGoogleCallback = async (req: Request, res: Response) => {
...
+        }
+
+        const email = profile.emails[0].value;
+
+        const newUserAttributes: Partial<UserAttributes> = {
+          name: `${profile.name?.givenName} ${profile.name?.familyName}` || '',
+          googleId: profile.id,
+          photoUrl: profile.photos?.[0]?.value || '',
+          email,
+          roleId: 'd290f1ee-6c54-4b01-90e6-d701748f0852', // Default roleId
```
Comment (fab-ryan) [inline src/controllers/authController.ts]: Declare this variable in external space

**Review by fab-ryan — APPROVED** [review]
Work on those requests

**Review by Nadinefiona — APPROVED** [review]

**Review by Ruthuwamahoro — APPROVED** [review]

**Pull request merged**